src/pages/panel/views/main.js

```diff
@@ -98,6 +105,32 @@ async function togglePause(host, event) {
   `);
 }
 
+async function toggleZapped(host) {
+  const { options, stats, paused } = host;
+
+  const zappedHostname = getTopDomainFromRecord(options.zapped, stats.hostname);
+
+  await store.set(options, {
```

Comment:
Best to move business logic to the background, but if you rely on some side effects of having it in the view, then at least, please encapsulate it in a function and preferably move to non-view file.